addons/crm_iap_lead/models/crm_iap_lead_mining_request.py

```diff
@@ -119,6 +122,33 @@ def _compute_team_id(self):
             team = self.env['crm.team']._get_default_team_id(user_id=user.id, domain=team_domain)
             mining.team_id = team.id
 
+    @api.depends('country_ids')
+    def _compute_available_state_ids(self):
+        """ States for some specific countries should not be offered as filtering options because
+        they drastically reduce the amount of IAP reveal results.
+
+        For example, in the United States, only ~64.000 companies have a defined state within the
+        reveal service while ~717.770 companies have no state defined at all.
+
+        Meaning specifying states for that country will yield a lot less results than what you could
+        expect, which is not the desired behavior.
+        Obviously all companies are active within a state, it's just a lack of data in the reveal
+        service side.
+
+        To help users create meaningful iap searches, we remove the states filtering for several
+        blacklisted countries (based on their country code).
+        The complete list and reasons for this change can be found on task-2471703. """
+
+        blacklisted_country_codes = ['BS', 'BE', 'CL', 'CN', 'CO', 'CR', 'CU', 'CG', 'DO', 'EC',
```

Comment:
Can't we move that (+ a small explanation coming from the famous task-2471703) in a global variable available as an IAP tool somewhere ? A bit like blacklisted mail providers, ...

Reply:
Moved to iap_tools next to the mailing blacklist.